lib/puppetfile-resolver/puppetfile/git_module.rb

```diff
@@ -10,10 +10,89 @@ class GitModule < BaseModule
       attr_accessor :commit
       attr_accessor :tag
 
+      alias :url :remote
+
+      REMOTE_REFS_CMD = 'git ls-remote --symref'
```

Comment:
I wonder, is it worth checking if `git` is even available?  Not everyone may have it on the PATH.

Reply:
If this is the case I would argue for some util libraries for running commands. Then those could search for git and possibly raise an error if not found.